File: web/frontend/src/domain/entities.ts
/**
 * Entidades del dominio - Representan los conceptos del negocio
 */

export interface Prediction {
  class: string;
  display_name: string;
  emoji: string;
  confidence: number;
}

export interface AllPrediction {
  class: string;
  display_name: string;
  emoji: string;
  confidence: number;
}

export interface PredictionResponse {
  success: boolean;
  prediction: Prediction;
  all_predictions: AllPrediction[];
}

export interface ImageClass {
  name: string;
  display_name: string;
  emoji: string;
}

export interface BatchResultItem {
  filename: string;
  success: boolean;
  prediction?: Prediction;
  all_predictions?: AllPrediction[];
  error?: string;
}

export interface BatchResponse {
  success: boolean;
  total: number;
  processed: number;
  failed: number;
  results: BatchResultItem[];
}

File: web/backend/app/domain/entities.py
"""
Entidades del dominio - Representan los conceptos del negocio
"""
from dataclasses import dataclass
from typing import List


@dataclass
class Prediction:
    """Resultado de una predicción del modelo"""
    class_name: str
    confidence: float
    all_predictions: List[dict]


@dataclass
class ImageClass:
    """Clase de imagen que puede ser clasificada"""
    name: str
    display_name: str
    emoji: str


# Clases disponibles en el modelo
CLASSES = [
    ImageClass("cloudy", "Nubes", "☁️"),
    ImageClass("desert", "Desierto", "🏜️"),
    ImageClass("green_area", "Área Verde", "🌿"),
    ImageClass("water", "Agua", "💧"),
]

